roles/acm_setup/tasks/main.yml

```
@@ -188,6 +195,8 @@
         name: provisioning-configuration
       spec:
         watchAllNamespaces: true
+  when:
+    - _as_prov_conf.resources | default([0]) | length == 1
```

This condition is met when there are no `Provisioning` resources, did you mean:
```suggestion
    - _as_prov_conf.resources | default([]) | length == 1
```